src/components/JobCard/JobCard.tsx

```diff
@@ -12,20 +12,14 @@ interface JobCardProps {
   job: Job;
   idx: number;
   pending: boolean;
-  pendingJobs: Job[];
-  setPendingJobs: React.Dispatch<React.SetStateAction<Job[]>>;
-  filteredJobs: Job[];
-  setFilteredJobs: React.Dispatch<React.SetStateAction<Job[]>>;
+  filterList: (idx: number) => void;
```

Comment:
Since this is a callback, I'd recommend falling the `on{ACTION}` convention by naming this `onRemove`.

Calling it `filterList` makes it seem like there's a list contained in `JobCard` and that function is used to filter it.